sourcegraph/sourcegraph
# unify search UI: remove local def search, add 's' hotkey for global search, repo homepage w/frequently used defs

The directory tree listing for repos no longer searches or lists defs.
Instead, you can open the global search modal anywhere by hitting the
`s` hotkey. This removes some functionality (local def searching, and
seeing defs on the repo homepage), but it will be added back shortly. 

Greatly improves the performance of the file finder and
branch/tag menu filtering (instead of using Fuze.js, now uses a
simpler fuzzysearch lib).

Adds a "Frequently used" list of defs to the repo homepage. After removing the defs from the TreeSearch, it's important to show some defs on the repo homepage so that users know there is more than just files.

Test plan:

https://staging4.sourcegraph.com

1. Go to a never-seen-before repo and ensure it clones properly and defs show up without refreshing the browser.
2. Use global search (hotkey: `s`) from many different pages.
3. Use TreeSearch for jump-to-file from many different pages.
4. Use the RevSwitcher.

NOTE: The repo homepage and the global search modal are not polished. These are not announced features/screens, so this is OK.

#### bare repo TreeSearch (w/no defs)
![screenshot from 2016-06-14 02-34-59](https://cloud.githubusercontent.com/assets/1976/16038179/be0db244-31d8-11e6-92e6-390ad1d97245.png)

#### global search modal (accessible using `s` hotkey from anywhere)
![screenshot from 2016-06-14 02-34-46](https://cloud.githubusercontent.com/assets/1976/16038178/be043386-31d8-11e6-8fe9-3f3668c7c133.png)

#### new repo homepage (not polished, but no worse than current homepage)
![screenshot from 2016-06-14 02-34-26](https://cloud.githubusercontent.com/assets/1976/16038177/bdf983c8-31d8-11e6-8097-a5084ed785a4.png)
